File: Compilers-II/Mini-Asgn-1_CS18BTECH11042/q1_t.py
# Simple program to demonstrate how Python interpreter allows for dynamic types, and how variables can be declared without type being specified.
# Shreyas Havaldar, CS18BTECH11042
t = 2
print(t)
t = "Hello"
print(t)

# The variable t, whose type is not specified is first an integer with value 2, and is thus printed as 2 on the console. Later the same variable is assigned value "Hello", which is an string, thus the same variable is used for representing two data types in the same program. The data type it is representing need not even be explicitly specified.
# Command: python q1_t.py 
# Output:
# 2
# Hello

File: Compilers-II/q1_t.c
// Simple program to demonstrate how GCC compiler does not allow for declaring variables without type being specified.
// Shreyas Havaldar, CS18BTECH11042

#include<stdio.h>

int main(){

    t = 2;
    printf("%d\n",t);
    t = "Hello";
    printf("%s\n",t);

    return 0;
}
// Thus we can see that the program does not get compiled at all and throws errors for type information.
// Command: gcc q1_t.c
// Output:
// q1_t.c: In function ‘main’:
// q1_t.c:8:5: error: ‘t’ undeclared (first use in this function)
//      t = 2;
//      ^
// q1_t.c:8:5: note: each undeclared identifier is reported only once for each function it appears in